api/v1alpha1/bundleinstance_types.go

```diff
@@ -46,25 +47,37 @@ const (
 type BundleInstanceSpec struct {
 	// ProvisionerClassName sets the name of the provisioner that should reconcile this BundleInstance.
 	ProvisionerClassName string `json:"provisionerClassName"`
+	// Label selector for Bundles resources
+	Selector *metav1.LabelSelector `json:"selector"`
+	// Template describes the generated Bundle that this instance will manage.
+	Template *BundleTemplate `json:"template"`
+}
 
-	// BundleName is the name of the bundle that this instance is managing on the cluster.
-	BundleName string `json:"bundleName"`
+// BundleTemplate defines the desired state of a Bundle resource
+type BundleTemplate struct {
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	// +kubebuilder:pruning:PreserveUnknownFields
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Specification of the desired behavior of the Bundle.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +optional
+	Spec BundleSpec `json:"spec,omitempty"`
 }
 
 // BundleInstanceStatus defines the observed state of BundleInstance
 type BundleInstanceStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-	Conditions []metav1.Condition `json:"conditions,omitempty"`
-
-	InstalledBundleName string `json:"installedBundleName,omitempty"`
+	Conditions         []metav1.Condition      `json:"conditions,omitempty"`
+	InstalledBundleRef *corev1.ObjectReference `json:"installedBundleRef,omitempty"`
+	DesiredBundleRef   *corev1.ObjectReference `json:"desiredBundleRef,omitempty"`
```

Comment:
TODO: figure out whether this is the correct implementation.

Reply:
Nope - this was more-or-less a stale artifact when I was throwing code against the wall last week. I'm still trying to work out in my head whether this kind of field is needed, or whether we still need to surface the desired bundle printer column in the immediate term.